src/retrieve-data-helpers/smart-launch.js

```diff
@@ -0,0 +1,48 @@
+import { signalSuccessFhirServerRetrieval } from '../actions/fhir-server-actions';
+import { signalSuccessSmartAuth, signalFailureSmartAuth } from '../actions/smart-auth-actions';
+import store from '../store/store';
+
+/**
+ * Handle SMART launch (Oauth2 response) accordingly if Sandbox is launched as a SMART on FHIR app.
+ * Store relevant data passed in.
+ * @returns {Promise} - Promise to resolve elsewhere
+ */
+function smartLaunchPromise() {
+  return new Promise((resolve, reject) => {
+    const onAuthFailure = () => {
+      console.log('Launched openly and unsecured');
+      store.dispatch(signalFailureSmartAuth());
+      return reject();
+    };
+
+    const onAuthSuccess = (smart) => {
+      console.log('Launched as a SMART app');
+      if (smart.tokenResponse) {
+        let fhirBaseUrl;
+        if (smart.server && smart.server.serviceUrl) {
+          fhirBaseUrl = smart.server.serviceUrl;
+          const conformanceQuery = smart.api.conformance({});
+          const deferredToPromise = Promise.resolve(conformanceQuery);
+          deferredToPromise.then((result) => {
+            store.dispatch(signalSuccessSmartAuth(smart));
+            store.dispatch(signalSuccessFhirServerRetrieval(fhirBaseUrl, result.data));
```

Comment:
Would success on getting Fhir Server data indicate a successful smart auth here ? 
If so you could reduce these dispatches to single call taking in a hash in turn updating both SMART_AUTH_SUCCESS and GET_FHIR_SERVER_SUCCESS.

Reply:
Good point. I initially thought it would be better to separate the concerns of storing FHIR Metadata vs storing SMART authentication data, but I guess there's no problem with combining. 
https://github.com/cds-hooks/sandbox-2.0/pull/2/commits/edd5c152ce616fb85dd02eea0014873ced111905#diff-4a1f10861c6f9fe4853c7a307042c63eR24